### PR TITLE
Cert passphrase

### PR DESCRIPTION
**Pull request by debonair**
Allows you to upload certificates that have password protectected keys

### CODE REVIEW

**Comment by debonair** [comment]
This is waiting on this https://github.com/andris9/pem/pull/53 the test should pass when the code is merged in

**Comment by debonair** [comment]
@rcrichton my pem pull request has finally been merged in and this build passes now may I merge this in?

File: src/api/keystore.coffee [hunk 1]
```diff
@@ -168,7 +188,7 @@ exports.getCertKeyStatus = getCertKeyStatus = (callback) ->
   Keystore.findOne (err, keystoreDoc) ->
     return callback err, null if err
 
-    pem.getModulus keystoreDoc.key, (err, keyModulus) ->
+    pem.getModulusFromProtected keystoreDoc.key, keystoreDoc.passphrase, (err, keyModulus) ->
```
Comment (rcrichton) [inline src/api/keystore.coffee]: What happens if you call this with a key that isn't protected, will it still work?

Comment (debonair) [inline src/api/keystore.coffee]: Nothing.. it ignores the password part and works

File: src/koaApi.coffee [hunk 2]
```diff
@@ -103,6 +103,7 @@ exports.setupApp = (done) ->
   app.use route.post '/keystore/key', keystore.setServerKey
   app.use route.post '/keystore/ca/cert', keystore.addTrustedCert
   app.use route.get '/keystore/validity', keystore.verifyServerKeys
+  app.use route.post '/keystore/passphrase', keystore.setServerPassphrase
```
Comment (rcrichton) [inline src/koaApi.coffee]: Do we need a separate API request for just setting the passphrase? I see this is happening in the serServerCert call to. Maybe we can just do it there?

Comment (debonair) [inline src/koaApi.coffee]: You may want to update just the passphrase without updating the key

File: src/api/keystore.coffee (hunk 1 above)
Comment (rcrichton) [inline src/api/keystore.coffee]: Ok cool

**Comment by rcrichton** [comment]
Hey @debonair this looks good. Just a few comments above. Also, could you add a test to ensure that the server will start up with a protected key. Just for future safety.

File: src/koaApi.coffee (hunk 2 above)
Comment (rcrichton) [inline src/koaApi.coffee]: Ok, I don't see a problem having both. Is this used in the console atm?

Comment (debonair) [inline src/koaApi.coffee]: yes it is, will push the console pull request just now.

**Comment by rcrichton** [comment]
Hey @debonair, please wait for a thumbsup before merging in so that I have a chance to see the changes you've made. This does looks good so it's ok this time around.

**Comment by debonair** [comment]
closes #465 #417

**Pull request merged**